OSDOCS-12261 autorepair setting for machine pools (#90634)
openshift/openshift-docs: modules/rosa-configuring-autorepair.adoc

// Module included in the following assemblies:
//
// * rosa_cluster_admin/rosa_nodes/rosa-managing-worker-nodes.adoc
// * nodes/rosa-managing-worker-nodes.adoc
//

:_mod-docs-content-type: PROCEDURE
[id="rosa-configuring-autorepair_{context}"]
= Configuring machine pool AutoRepair

> {hcp-title} supports an automatic repair process for machine pools. AutoRepair is useful when you would want the ROSA service to detect certain unhealthy machines, drain the unhealthy nodes, and re-create the nodes. You can disable auto-repair if the unhealthy machines should not be replaced in cases where the nodes or the unhealthy instances should be preserved. AutoRepair is enabled by default on Machine Pools. 

Reviewer: automatic repair, AutoRepair, auto-repair. 
Should we settle on one? 
```suggestion
{hcp-title} supports an automatic repair process for machine pools, called AutoRepair. AutoRepair is useful when you want the ROSA service to detect certain unhealthy machines, drain the unhealthy nodes, and re-create the nodes. You can disable AutoRepair if the unhealthy machines should not be replaced, such as in cases where the nodes or the unhealthy instances should be preserved. AutoRepair is enabled by default on Machine Pools. 
```
Also, machines, nodes, instances. Are these three different concepts, or different words for the same thing?

Author: Same thing, I will make it consistent

The AutoRepair process deems a node unhealthy when the State of the node is either `NotReady` or Unknown for predefined amount of time (typically 8 minutes). Whenever two or more nodes become unhealthy simultaneously, the AutoRepair process will stop repairing the nodes.
Similarly, when a new node is created unhealthy even after a predefined amount of time (typically 20 minutes), the service will auto-repair. 

[NOTE]
====

> Machine pool AutoRepair is only available for {hcp-title} clusters at this time. 

Reviewer: I think we tend to avoid wording that suggests a new feature is coming (in case it doesn't). Also, this 
```suggestion
Machine pool AutoRepair is only available for {hcp-title} clusters . 
```
